public/store/reducers/workspace_reducer.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  isDirty: false,
+  components: [],
+};
+
+const workspaceSlice = createSlice({
+  name: 'workspace',
+  initialState,
+  reducers: {
+    setDirty(state) {
+      state.isDirty = true;
+    },
+    removeDirty(state) {
+      state.isDirty = false;
+    },
```

Comment:
We can - prefer to leave for now and as more implementation comes we can decide on if we want a pattern like this, or what you're suggesting. It may get complex where one action may be updating several different state vars.